frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js

```diff
@@ -298,6 +298,34 @@ describe("scenarios > admin > settings", () => {
     });
   }
 
+  it("'General' admin settings should handle setup via `MB_SITE_ULR` environment variable (metabase#14900)", () => {
+    cy.server();
+    // 1. Get the array of ALL available settings
+    cy.request("GET", "/api/setting").then(({ body }) => {
+      // 2. Create a stubbed version of that array by passing modified "site-url" settings
+      const STUBBED_BODY = body.map(setting => {
+        if (setting.key === "site-url") {
+          const STUBBED_SITE_URL = Object.assign({}, setting, {
+            is_env_setting: true,
+            value: null,
+          });
+
+          return STUBBED_SITE_URL;
+        }
+        return setting;
+      });
+
+      // 3. Stub the whole response
+      cy.route("GET", "/api/setting", STUBBED_BODY).as("appSettings");
```

Comment:
Before we get access to `intercept`, this double-GET approach is great! It could even be extracted pretty cleanly if we do a lot of it.